File: 3778-transform-array-by-parity/transform-array-by-parity.cpp
class Solution {
public:
    vector<int> transformArray(vector<int>& nums) {
        int j = 0;
        for(auto i : nums){
            if(!(i&1))
                nums[j++] = 0;
            else nums[j++] = 1;
        }
         sort(nums.begin(),nums.end());
        return nums;
    }
};

File: 3778-transform-array-by-parity/transform-array-by-parity.java
class Solution {
    public int[] transformArray(int[] nums) {
        int j = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] % 2 == 0) {
                nums[j++] = 0;
            }
        }
        while (j < nums.length) {
            nums[j++] = 1;
        }
        return nums;
    }
}